### PR TITLE
Add helpful parameters to `generate_freq_group_membership_array` and `compute_freq_by_strata`

### PR DESCRIPTION
**Pull request by jkgoodrich**
Needed for changes in gnomad_qc [PR](https://github.com/broadinstitute/gnomad_qc/pull/450)

### CODE REVIEW

File: gnomad/utils/annotations.py [hunk 1]
```diff
@@ -1736,6 +1758,9 @@ def compute_freq_by_strata(
         function.
     :param select_fields: Optional list of row fields from `mt` to keep on the output
         Table.
+    :param group_membership_includes_raw_group: Whether the 'group_membership'
```
Comment (KoalaQin) [inline gnomad/utils/annotations.py:1761]: Maybe add explanation on "inserting" and "changing"

**Review by KoalaQin — CHANGES_REQUESTED** [review]

File: gnomad/utils/annotations.py [hunk 2]
```diff
@@ -1783,10 +1808,17 @@ def _agg_by_group(
...
             )
         )
+        # Create final agg list by inserting or changing the "raw" group,
+        # representing all samples, in the adj_agg_list.
         raw_agg_expr = ann_expr.aggregate(lambda x: agg_func(x, *args))
-        # Create final agg list by inserting the "raw" group, representing all samples,
-        # into the adj_agg_list.
-        return adj_agg_expr[:1].append(raw_agg_expr).extend(adj_agg_expr[1:])
+        if group_membership_includes_raw_group:
+            adj_agg_expr[1] = raw_agg_expr
```
Comment (KoalaQin) [inline gnomad/utils/annotations.py:1815]: I ran into error of this line while testing: 
`TypeError: 'ArrayStructExpression' object does not support item assignment`

Comment (jkgoodrich) [reply gnomad/utils/annotations.py]: Sorry, some of my changes didn't get committed, I added them, so you shouldn't get this error anymore

**Review by KoalaQin — CHANGES_REQUESTED** [review]

File: gnomad/utils/annotations.py [hunk 3]
```diff
@@ -1680,6 +1687,12 @@ def generate_freq_group_membership_array(
         [hl.agg.count_where(x[1]) for x in sample_group_filters]
     )
 
+    if remove_zero_sample_groups:
+        filter_freq = hl.enumerate(freq_meta_sample_count).filter(lambda x: x[0] > 0)
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad/utils/annotations.py:1691]: nope, you are right, thanks!

**Review by KoalaQin — CHANGES_REQUESTED** [review]

**Review by KoalaQin — APPROVED** [review]
LGTM!

**Pull request merged**